imports/plugins/core/core/server/util/getAbsoluteUrl.js

```diff
@@ -0,0 +1,18 @@
+/**
+ * @name getAbsoluteUrl
+ * @summary Returns the absolute/base URL, returning process.env.ROOT_URL if set,
+ *  otherwise using the request's protocol & hostname
+ * @param {Object} request Express request object
+ * @param {Object} request.hostname Hostname derived from Host or X-Forwarded-Host heaer
+ * @param {String} request.protocol Either http or https
+ * @returns {String} URL
+ */
+export default function getAbsoluteUrl(request) {
```

Comment:
Similar to how the current `Reaction.absoluteUrl` function works, this should take an optional `path` arg and then do the work of properly appending that path. It should check if it starts with `/` and remove it so that we don't end up with two `/`.

In your xform you will then do something like `large: context.getAbsoluteUrl(large)`.

Actually, performance wise, it's best if you have two different functions. This function you have now would be called only one time in `buildContext`. You could actually set the result of that on context, too, as `rootUrl`. Then `context.getAbsoluteUrl` would only do the work to combine root url with path. Something like:

```js
function buildContext() {
  context.rootUrl = getRootUrl(request); // this is the function you currently have, renamed
  context.getAbsoluteUrl = (path) => getAbsoluteUrl(context.rootUrl, path); // this just combines the strings intelligently
}
```